var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -349,6 +353,11 @@ def setup_environment(self, spack_env, run_env):
         run_env.set('PETSC_DIR', self.prefix)
         run_env.unset('PETSC_ARCH')
 
+        # Petsc build will use python2 internaly
```

Comment:
Could you please detail a little bit this comment? It is not very clear to me.

Reply:
If you look at [this](https://bitbucket.org/petsc/petsc/commits/a3e07f7d98935102301a2849d6c5439db2167066), this is what happens : 

```
-import sys
+import sys, os
 if not type(sys.version_info) is tuple and sys.version_info.major > 2:
-  print('Configure does not support Python 3 yet, please run as')
+  if os.path.basename(sys.executable) == 'python2':
+    # Exit to prevent an infinite loop in case the environment is corrupted such
+    # that "python2" in PATH is actually a Python 3 interpreter.
+    print('Executable not a valid Python 2 interpreter: ' + sys.executable)
+    sys.exit(1)
+  print('Configure does not support Python 3 yet, attempting to run as')
   print('  python2 ' + ' '.join(["'" + a + "'" for a in sys.argv]))
-  sys.exit(1)
+  os.execlp('python2', 'python2', *sys.argv)
```

This is hack but currently there is no other way to have Python3 in the dependency tree when PETSc is involved as one of the dependency .